From 9377d7e9521650c222dce5c55625b4e923af7600 Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Tue, 29 Sep 2009 22:26:22 +0300
Subject: Work around an ICE on ia64

Bug-Debian: http://bugs.debian.org/548943


Temporarily work around an internal compiler error in Devel::PPPort
on ia64+gcc-4.3.


Patch-Name: debian/devel-ppport-ia64-optim.diff
---
 cpan/Devel-PPPort/Makefile.PL |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/cpan/Devel-PPPort/Makefile.PL b/cpan/Devel-PPPort/Makefile.PL
index 67eebc1..f1ef7a2 100644
--- a/cpan/Devel-PPPort/Makefile.PL
+++ b/cpan/Devel-PPPort/Makefile.PL
@@ -75,6 +75,13 @@ sub configure
     push @moreopts, INSTALLDIRS => ($] >= 5.007003 ? 'perl' : 'site');
   }
 
+
+  # temporary Debian hack, see http://bugs.debian.org/548943
+  require Config;
+  if ($Config::Config{archname} =~ /^ia64/) {
+    push @moreopts, OPTIMIZE => '-g -O0';
+  }
+
   if ($opt{'apicheck'}) {
     $PL_FILES{'apicheck_c.PL'} = 'apicheck.c';
     push @C_FILES, qw{ apicheck.c };
-- 
1.7.5.1

